third_party/proto/cosmos/base/coin.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+package cosmos.base.v1beta1;
+
+import "gogoproto/gogo.proto";
+
+option go_package                       = "github.com/cosmos/cosmos-sdk/types";
+option (gogoproto.goproto_stringer_all) = false;
+option (gogoproto.stringer_all)         = false;
+
+// Coin defines a token with a denomination and an amount.
+//
+// NOTE: The amount field is an Int which implements the custom method
+// signatures required by gogoproto.
+message Coin {
+  option (gogoproto.equal) = true;
+
+  string denom  = 1;
+  string amount = 2 [(gogoproto.customtype) = "Int", (gogoproto.nullable) = false];
```

Comment:
The type we use for amount is different than the amount in message MsgLock?

MsgLock is 
```
string amount = 2 [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
```

Reply:
This is an internal cosmos proto implementing Coin type, so it looks different.